pkg/parser/uri_test.go

```diff
@@ -0,0 +1,68 @@
+package parser
+
+import (
+	"testing"
+)
+
+type tests struct {
+	dsn    string
+	scheme string
+	path   string
+	query  string
+}
+
+var fTests = []tests{
+	{"", "", "", ""},
```

Comment:
This looks like a invalid url and should probably not be in this list.

Reply:
I was thinking about the zero-boundary of the tests. I don't know if this will ever happen, but in case we get an empty string, we want to catch it and return an error. That's the idea behind the test case. I've moved it to a different test function – `TestParseURIErrors`